FinalChallenge/Views/AllTasks/AllTasksViewModel.swift

```
@@ -0,0 +1,65 @@
+//
+//  AllTasksViewModel.swift
+//  FinalChallenge
+//
+//  Created by aluno on 15/10/19.
+//  Copyright © 2019 Elias Paulino. All rights reserved.
+//
+import CoreData
+import UIKit
+
+struct AllTasksViewModel<FrequentRepository: RepositoryErasurer<FrequencyTask, NSManagedObjectID>,
+                         ZoneRepository: RepositoryErasurer<ZoneTask, NSManagedObjectID>> {
+    
+    private var frequentRepository: FrequentRepository
+    private var zoneRepository: ZoneRepository
+    
+    init(frequentRepository: FrequentRepository, zoneRepository: ZoneRepository) {
+        self.frequentRepository = frequentRepository
+        self.zoneRepository = zoneRepository
+    }
+    
+    lazy var tasks: ([FrequencyTask], [ZoneTask]) = {
+        let frequencyTasks = readAllFrequencyTasks()
+        let zoneTasks = readAllZoneTasks()
+        let tasks = (frequencyTasks, zoneTasks)
+        return tasks
+    }()
+    
+    lazy var numberOfRows: Int = {
+        let rows = (tasks.0.count) + (tasks.1.count)
+        return rows
+    }()
+    
+    func readAllFrequencyTasks() -> [FrequencyTask] {
+        var tasks = [FrequencyTask]()
+        
+        frequentRepository.readAll { result in
+            switch result {
+            case .error(let error):
+                print(error)
+                fatalError("error no frenquent repository")
+            case .item(let task):
+                tasks.append(contentsOf: task)
+            }
+        }
+        
+        return tasks
+    }
+    
+    func readAllZoneTasks() -> [ZoneTask] {
+        var tasks = [ZoneTask]()
+
+        zoneRepository.readAll { result in
+            switch result {
+            case .error(let error):
+                print(error)
+                fatalError("error no zone repository")
```

should throw a fatal error?

---

ok